src/main/java/funda/prog101/MarkIV.java

```diff
@@ -67,10 +72,32 @@ public MarkIV() {
 	/**
 	 * This method is for setting startButton
 	 * @param switchButton boolean parameter;
+	 * @throws InterruptedException 
 	 */
-	public void powerButton(boolean switchButton) {
+	public void powerButton(boolean switchButton) throws InterruptedException {
 		this.switchButton = switchButton;
-		this.light = switchButton;
+		if (switchButton) {
+			heaterBoiler.switchHeater();
+			boiler.switchPipe();
+			Thread.sleep(10000);
+			this.light = true;
+			for (int item = 0; item < boiler.limit; item++) {
+				boiler.quantity--;
+				coffeeFilter.addCups();
```

Comment:
It is not necessary this code line, you have initiated variables in CoffeeFilter builder